src/components/bar.cpp

```diff
@@ -546,6 +548,16 @@ void bar::reconfigure_pos() {
  * Reconfigure window strut values
  */
 void bar::reconfigure_struts() {
+  if (!m_opts.struts) {
+    if (m_conf.has("global/wm", "margin-bottom")) {
+      m_log.warn("Struts are disabled, ignoring margin-bottom");
+    }
+    if (m_conf.has("global/wm", "margin-top")) {
+      m_log.warn("Struts are disabled, ignoring margin-top");
+    }
+    return;
+  }
+
```

Comment:
Like this, the warning gets printed every time the function is called (this happens when you toggle the bar). It would be better to move this to where the config options are loaded